File: algorithms/ar-expsq/python2/exponentiation_by_squaring.py
def exponentiationSquaring(base, power):
    result = 1
    while(power > 0):
        if power % 2 != 0:  # if power is ODD multiply result by base
            result = (result * base)

        base = base * base  # squaring base
        power /= 2  # half the power

    return result


def main():
    power = 3
    base = 5
    result = exponentiationSquaring(base, power)
    print(result)


if __name__ == '__main__':
    main()

File: algorithms/ar-expsq/c/exponentiation_by_squaring.c
#include <stdio.h>
#include <stdlib.h>

void exponentiationSquaring(long base, long power) {
	long result = 1;

	while(power > 0) { //run loop till power is greater than zero
		if(power % 2 != 0) { //if power is ODD multiply result by base
			result = result * base;
		}
		base = base * base; //squaring base
		power /= 2; //reduce power to half
	}
	
	printf("%li\n", result);
}

int main() {
	long base = 3;
	long power = 2;

	exponentiationSquaring(base, power);
	return 0;
}
